plugins/aws/s3/src/hopeit/aws/s3/object_storage.py

```diff
@@ -128,12 +122,12 @@ def __init__(
         :param prefix, Optional[str]: Prefix to be used for every element (object or file) stored in the S3 bucket.
         :param partition_dateformat, Optional[str]: Optional format string for partitioning
             dates in the S3 bucket.
-        :param create_bucket, bool: Whether to create the S3 bucket if it doesn't exist
         """
         self.bucket: str = bucket
-        self.prefix: Optional[str] = prefix
+        self.prefix: Optional[str] = (
+            (prefix.rstrip("/") + "/") if prefix and prefix != "/" else None
```

Comment:
no need to check prefix != '/', if the backed does not support it it should fail.